tests/BuildScriptGenerator.Tests/Node/NodePackageScriptsCheckerTest.cs

```diff
@@ -0,0 +1,49 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.Oryx.BuildScriptGenerator.Node;
+using Microsoft.Extensions.Logging.Abstractions;
+using Xunit;
+using Microsoft.Oryx.Tests.Common;
+using Microsoft.Oryx.Common;
+
+namespace Microsoft.Oryx.BuildScriptGenerator.Tests.Node
+{
+    public class NodePackageScriptsCheckerTest
+    {
+        [Fact]
+        public void Checker_DetectsGlobalNpmInstalls()
+        {
+            // Arrange
+            var checker = new NodePackageScriptsChecker(null);
+            var scripts = new Dictionary<string, string>
+            {
+                { "preinstall",     "npm i pkg -g # checked and problematic" },
+                { "install",        "echo bla bla # checked and not problematic" },
+                { "postshrinkwrap", "npm i -g pkg # not checked" }
+            };
+
+            // Act & Assert
+            Assert.Single(checker.CheckScriptsForGlobalInstallationAttempts(scripts));
+        }
+
+        [Theory]
+        [InlineData("npm install -g pkg",       true)]
+        [InlineData("npm install --global pkg", true)]
+        [InlineData("npm i -g pkg",             true)]
+        [InlineData("npm i --global pkg",       true)]
+        [InlineData("npm install pkg",                  false)]
```

Comment:
"npm install\ngrep -g bla"

Reply:
scripts in `package.json` can't span multiple lines